src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -139,34 +141,45 @@ public void onTableRemoved(final TableMetadata table) {
                     final String ksName = table.getKeyspace().getName();
                     final String tableName = table.getName();
 
-                    final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(table.getKeyspace().getName());
+                    final org.apache.cassandra.schema.KeyspaceMetadata oldKsm = Schema.instance.getKSMetaData(ksName);
 
                     if (oldKsm == null) {
-                        LOGGER.warn("KeyspaceMetadata for keyspace {} is not found!", table.getKeyspace().getName());
+                        LOGGER.warn("KeyspaceMetadata for keyspace {} is not found!", ksName);
                         return;
                     }
 
-                    final ColumnFamilyStore cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+                    ColumnFamilyStore cfs;
 
-                    if (cfs == null) {
-                        LOGGER.warn("ColumnFamilyStore for {}.{} is not found!", table.getKeyspace(), table.getName());
+                    try {
+                        cfs = Keyspace.openWithoutSSTables(ksName).getColumnFamilyStore(tableName);
+                    }
+                    catch (final Exception ex) {
+                        LOGGER.warn("ColumnFamilyStore for {}.{} is not found!", ksName, table.getName());
```

Comment:
In which situation would this happen? A comment in the code would be great.

Reply:
Right now, after we fixed it, we should be pretty safe to never encounter this again. This is more to be just sure to log it but I cant see how it might happen again.

Reply:
Ok, thx.